build_runner/test/daemon/daemon_test.dart

```diff
@@ -240,6 +252,53 @@ main() {
       expect(ddcContent, contains('goodbye world'));
     });
 
+    test('step build mode does not automatically build without continuing',
+        () async {
+      await _startDaemon(buildMode: BuildMode.Step);
+      var client = await _startClient(buildMode: BuildMode.Step)
+        ..registerBuildTarget(webTarget);
+      clients.add(client);
+      // Let the target request propagate.
+      await Future<void>.delayed(Duration(seconds: 2));
```

Comment:
😭 does `_startClient` need to be updated? Maybe we need a `registerSuccess` response to let us know the build daemon is ready? These types of artifical delays need to be a last resort.

That doesn't have to happen in this PR but it would be a good general change I think and help eliminate real world race conditions as well.

Reply:
That could help although this only shows up in testing. I can actually rework this test to not require the delay.

Reply:
If you can rework it to not require the delay please do.